lib/ansible/modules/web_infrastructure/jenkins_job.py

```diff
@@ -208,31 +208,31 @@ def get_jenkins_connection(self):
             else:
                 return jenkins.Jenkins(self.jenkins_url)
         except Exception as e:
-            self.module.fail_json(msg='Unable to connect to Jenkins server, %s' % to_native(e), exception=traceback.format_exc())
+            self.module.fail_json(msg='Unable to connect to Jenkins server, %s' % ensure_str(e), exception=traceback.format_exc())
 
     def get_job_status(self):
         try:
             response = self.server.get_job_info(self.name)
             if "color" not in response:
                 return self.EXCL_STATE
             else:
-                return response['color'].encode('utf-8')
+                return ensure_str(response['color'])
 
         except Exception as e:
-            self.module.fail_json(msg='Unable to fetch job information, %s' % to_native(e), exception=traceback.format_exc())
+            self.module.fail_json(msg='Unable to fetch job information, %s' % ensure_str(e), exception=traceback.format_exc())
 
     def job_exists(self):
         try:
             return bool(self.server.job_exists(self.name))
         except Exception as e:
-            self.module.fail_json(msg='Unable to validate if job exists, %s for %s' % (to_native(e), self.jenkins_url),
+            self.module.fail_json(msg='Unable to validate if job exists, %s for %s' % (ensure_str(e), self.jenkins_url),
                                   exception=traceback.format_exc())
 
     def get_config(self):
         return job_config_to_string(self.config)
 
     def get_current_config(self):
-        return job_config_to_string(self.server.get_job_config(self.name).encode('utf-8'))
+        return job_config_to_string(self.server.get_job_config(self.name))
```

Comment:
I like the fact that you move the potential conversion from text to bytes lower in the call stack, to where the API actually needs a byte string.